pkg/kube/wait.go

```diff
@@ -91,31 +87,31 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 				}
 				deployments = append(deployments, newDeployment)
 			case (*extensions.DaemonSet):
```

Comment:
Should we migrate [daemonsets](https://github.com/kubernetes/api/blob/81aa34336d28aadc3a8e8da7dfd9258c5157e5e4/apps/v1beta2/generated.proto#L67), [deployments](https://github.com/kubernetes/api/blob/81aa34336d28aadc3a8e8da7dfd9258c5157e5e4/apps/v1beta2/generated.proto#L196), [replicasets](https://github.com/kubernetes/api/blob/81aa34336d28aadc3a8e8da7dfd9258c5157e5e4/apps/v1beta2/generated.proto#L340) as well as [statefulsets](https://github.com/kubernetes/api/blob/81aa34336d28aadc3a8e8da7dfd9258c5157e5e4/apps/v1beta2/generated.proto#L558) to use apps/v1beta2? I see you've done that for StatefulSets, but is there something blocking us from doing the same for these resources as well?

I'm basing my knowledge from http://blog.kubernetes.io/2017/09/kubernetes-18-security-workloads-and.html if that makes any sense. I'm not sure if that'd break existing charts or not, but I assume this would be fixed now because of #2976.

Reply:
good question... spelunking

Reply:
I guess switching to `apps/v1beta2` would break backwards compatibility to older k8s versions, right? The dashboard developers are just using the old api: https://raw.githubusercontent.com/kubernetes/dashboard/master/src/deploy/recommended/kubernetes-dashboard.yaml

Reply:
Note, for 1.9 it will be `app/v1`. That change is already on master.

Reply:
Is the helm api philosophy to use an api supported by the N and N-1 versions of kubernetes?

Reply:
I included both types